In [2]:
import sys

import cobra.manipulation.delete
from cobra.io import read_sbml_model

sys.path.insert(0, r"C:\Users\Bisbii\PythonProjects\ExpAlgae\src")
from ExpGSMM.model.COBRAmodel import *
from cobra.flux_analysis import pfba, find_blocked_reactions, flux_variability_analysis as fva

In [143]:
data_directory = "../data"
dmodel = MyModel(join(data_directory, "models/model_dfba.xml"), "e_ActiveBiomass__cytop")

Loading



Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.


Reactions: 3512
Metabolites: 3660
Genes: 1690
Model loaded


In [7]:
from_carveme = read_sbml_model(r"C:\Users\Bisbii\Downloads\model_Mtuberculosis_carveme_carveme.xml")

In [8]:
print(len(from_carveme.reactions))
print(len(from_carveme.metabolites))
print(len(from_carveme.genes))
print(len(from_carveme.compartments))

1487
1087
944
3


In [7]:
from cobra.io import read_sbml_model, write_sbml_model
from_merlin = read_sbml_model(r"C:\Users\Bisbii\PythonProjects\ExpGSMM\data\models\model_carve.xml")

12345 does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id
No objective in listOfObjectives
No objective coefficients in model. Unclear what should be optimized


In [8]:
print(len(from_merlin.reactions))
print(len(from_merlin.metabolites))
print(len(from_merlin.genes))
print(len(from_merlin.compartments))

1487
1087
944
3


In [16]:
[''.join(e.id.split("_")[:-1]) for e in from_merlin.reactions]

['UAMAGS',
 'VCOAD',
 'HMPK1',
 'MDDCP4pp',
 'SAOR',
 'ARGDC',
 'GDPTPDP',
 'DRU1Ptex',
 'KSTD',
 'EXoxa',
 'FACOAL90t2pp',
 '2AGPEAT120',
 'MICITDr',
 'G6PP',
 'AHSERL2',
 'DHDPS',
 'MAN1PT2',
 'ECOAH37',
 'TSULabc',
 'EXcmcbtt',
 'CGLYabcpp',
 'EXbmn',
 'RU5PP',
 'MGt5',
 'RBFSa',
 '2AGPGAT181',
 'NDPK1',
 'RHACOARP100',
 'ARBt2rpp',
 'NTPP2',
 'ACTNabc1',
 'FCOAH2',
 'CDAPPA161',
 'GUAtex',
 'ACOAD24f',
 'ZNabc',
 'ASAD',
 'EXsalchs4',
 'ASPt2r',
 'DTMPK',
 'MTAN',
 'EXg3pe',
 'FADD3',
 'MCOATA',
 'NADH4',
 'FACOAL141t2pp',
 'ACACT1r',
 'TREabc',
 'AMALT3',
 'GLYCL',
 'CHRPL',
 'RECOAH20',
 'GLYALDDr',
 'GUAD',
 'RPI',
 'CAT',
 'NTPTP1',
 'FRD3',
 'GLUabc',
 'IMPC',
 'GALURt2r',
 'PG161abcpp',
 'ECHA192',
 'RHACOAR141d7',
 'PPTT',
 'GTHS',
 'EXcresol',
 'ICL',
 'EXalatrp',
 'LACZpp',
 'RHACOART60',
 'MLTP2',
 'EXgalur',
 'PG120abcpp',
 'DPPS',
 'VPAMTr',
 'HAT40tpp',
 'FACOAL120t2pp',
 'DASYN140',
 'PUNP7',
 'EXad',
 'AI2tpp',
 'PANTS',
 'GLYK',
 'GLYBabcpp',
 'SHCHD2',
 'NAabcO',
 

In [19]:
print(set([e.id for e in from_carveme.reactions]).difference(set([''.join(e.id.split("_")[:-1]) for e in from_merlin.reactions])))

{'EX_cgly_e', 'EX_cobalt2_e', 'EX_leu__L_e', 'EX_cytd_e', 'EX_ser__D_e', 'EX_pheme_e', 'EX_ival_e', 'EX_acysbmn_e', 'INOST_Et', 'EX_glyb_e', 'HBCO_nadp', 'DRIBtex_1', 'EX_stfrnA_e', 'EX_lac__L_e', 'EX_LalaDgluMdapDala_e', 'SPT_syn', 'EX_man6pglyc_e', 'EX_asp__L_e', 'HISTP_1', 'EX_h2o_e', 'MDH3_1', 'ARGDI_1', 'EX_spmd_e', 'L_LACt2r', 'EX_glcn_e', 'EX_gthrd_e', 'CHSTEROLt_1', 'EX_chol_e', 'EX_fru_e', 'EX_ser__L_e', 'FDMO_1', 'EX_3mb_e', 'EX_h2_e', 'EX_co_e', 'HMR_9638', 'EX_fe3dcit_e', 'EX_bzalc_e', 'EX_fe3_e', 'EX_cl_e', 'EX_csn_e', 'EX_s_e', 'PGMT_B', 'GTHRDt2_1', 'EX_oxa_e', 'EX_ala_L_glu__L_e', 'EX_tsul_e', 'EX_alatrp_e', 'CYTB_B3', 'EX_no2_e', 'NAR_syn_2', 'EX_phept_e', 'EX_pta_e', 'ALAD_L', 'EX_pro__L_e', 'EX_tartr__L_e', 'EX_so4_e', 'EX_ala_B_e', 'EX_nona_e', 'EX_ala_L_asp__L_e', 'LDH_L', 'EX_co2_e', 'SERD_D', 'EX_aso3_e', 'EX_indole_e', 'EX_ptrc_e', 'MNabc_rev', 'L_LACD3', 'EX_but_e', 'EX_met__D_e', 'EX_bmn_e', 'GALUR_Et', 'MS_1', 'EX_lcts_e', 'EX_anhgm_e', 'EX_pyr_e', 'sink_amob

In [5]:
from cobra.io import validate_sbml_model
validate_sbml_model(r"C:\Users\Bisbii\PythonProjects\ExpGSMM\data\models\model_carve.xml")

(<Model model_mtuberculosis_carveme at 0x1d363e22730>,
 {'SBML_FATAL': [],
  'SBML_ERROR': [],
  'SBML_SCHEMA_ERROR': [],
  'SBML_WARNING': [],
  'COBRA_FATAL': [],
  'COBRA_ERROR': ['No objective coefficients in model. Unclear what should be optimized'],
  'COBRA_WARNING': ["12345 does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id",
   'No objective in listOfObjectives'],
  'COBRA_CHECK': []})

In [100]:
from cobra.io import read_sbml_model, write_sbml_model
aragem = read_sbml_model(r"C:\Users\Bisbii\Desktop\fuckaragem.xml")
aragem.objective_direction = 'min'

In [102]:
df = pd.read_csv(r"C:\Users\Bisbii\Desktop\reactionConstraints.txt", sep="\t")
for reaction in df.iterrows():
    aragem.reactions.get_by_id(reaction[1]['reactionID'].replace("R_", "")).bounds = (reaction[1]['lb'], reaction[1]['ub'])

In [103]:
aragem.summary(pfba(aragem))

In [104]:
write_sbml_model(aragem, r"C:\Users\Bisbii\Desktop\aragem_photo.xml")

In [110]:
ara = read_sbml_model(r"C:\Users\Bisbii\Desktop\aragem_photo.xml")

In [111]:
ara.summary()

In [171]:
data_directory = "../data"
dmodel = MyModel(join(data_directory, "models/model_dfba.xml"), "e_ActiveBiomass__cytop")
def get_g_per_gDW(reaction):
    g_per_gDW = {}
    if not all(met.formula for met in reaction.reactants):
        raise Exception ("These metabolites have no formula: ", [e.id for e in reaction.metabolites if not e.formula])
    for met in reaction.reactants:
        g_per_gDW[met] = reaction.metabolites[met] * met.formula_weight / 1000
    total = abs(sum(g_per_gDW.values()))
    g_per_gDW_normalized = {key: value/total for key, value in g_per_gDW.items()}
    for met in reaction.reactants:
        reaction.subtract_metabolites({met: reaction.metabolites[met]})
    for met, stoichiometry in g_per_gDW_normalized.items():
        reaction.add_metabolites({met: stoichiometry / met.formula_weight * 1000})


def remove_metabolite_from_biomass(model, biomass_reaction, metabolites):
    for index, metabolite in enumerate(metabolites):
        if type(metabolite) == str:
            metabolites[index] = model.metabolites.get_by_id(metabolite)
    for metabolite in metabolites:
        biomass_reaction.subtract_metabolites({metabolite: biomass_reaction.metabolites[metabolite]})
    get_g_per_gDW(biomass_reaction)

column_a = {e: [dmodel.reactions.e_Pigment__chlo.metabolites[e]] for e in dmodel.reactions.e_Pigment__chlo.metabolites}
remove_metabolite_from_biomass(dmodel, dmodel.reactions.e_Pigment__chlo, ["C05306__chlo"])
for e in dmodel.reactions.e_Pigment__chlo.metabolites:
    column_a[e].append(dmodel.reactions.e_Pigment__chlo.metabolites[e])
df = pd.DataFrame.from_dict(column_a, orient='index', columns=["before", "after"])
df

Loading



Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.


Reactions: 3512
Metabolites: 3660
Genes: 1690
Model loaded


before     after
C08614__chlo    -0.867625 -0.889435
C05306__chlo    -0.027435       NaN
C08601__chlo    -0.076490 -0.078413
C08591__chlo    -0.135411 -0.138815
C08606__chlo    -0.002596 -0.002662
C05433__chlo    -0.020883 -0.021408
C05307__chlo    -0.009232 -0.009464
C20484__chlo    -0.369372 -0.378657
C05432__chlo    -0.000668 -0.000685
C08579__chlo    -0.001181 -0.001210
C06098__chlo    -0.013420 -0.013757
C02094__chlo    -0.200480 -0.205520
e_Pigment__chlo  1.000000  1.000000

before     after
C08614__chlo    -0.867625 -0.889435
C05306__chlo    -0.027435       NaN
C08601__chlo    -0.076490 -0.078413
C08591__chlo    -0.135411 -0.138815
C08606__chlo    -0.002596 -0.002662
C05433__chlo    -0.020883 -0.021408
C05307__chlo    -0.009232 -0.009464
C20484__chlo    -0.369372 -0.378657
C05432__chlo    -0.000668 -0.000685
C08579__chlo    -0.001181 -0.001210
C06098__chlo    -0.013420 -0.013757
C02094__chlo    -0.200480 -0.205520
e_Pigment__chlo  1.000000  1.000000

In [139]:
dmodel.reactions.e_Pigment__chlo.reactants

[]

In [25]:
aragem.metabolites.S_ATP_p.summary()

In [21]:
aragem.reactions.BIO_L.summary()

In [6]:
nacl_model = MyModel(join(data_directory, "omics/nacl_h2o2_sorb/nacl/Dsalina_nacl_Local2_2_4_4_fastcore_t2_8.xml"), "e_Biomass__cytop")

Loading



Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.


Reactions: 2268
Metabolites: 3360
Genes: 1521
Model loaded


In [12]:
nacl_model.summary()

Runnning pFBA


In [35]:
with dmodel:
    print(dmodel.summary())
    dmodel.demands.DM_C00404__vacu.bounds = (-0.005, 1000)
    dmodel.exchanges.EX_C00009__dra.bounds = (0, 1000)
    print(dmodel.summary())

Runnning pFBA


Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.
Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.


Objective
1.0 e_ActiveBiomass__cytop = 0.18193277555723875

Uptake
------
  Metabolite       Reaction      Flux  C-Number C-Flux
C00001__extr EX_C00001__dra     5.836         0  0.00%
C00009__extr EX_C00009__dra   0.05346         0  0.00%
C00011__extr EX_C00011__dra      8.21         1 99.58%
C00059__extr EX_C00059__dra   0.03554         0  0.00%
C00080__extr EX_C00080__dra   0.09017         0  0.00%
C00120__extr EX_C00120__dra  0.001289        10  0.16%
C00205__extr EX_C00205__dra     629.9         0  0.00%
C00244__extr EX_C00244__dra    0.8778         0  0.00%
C00378__extr EX_C00378__dra 0.0007412        12  0.11%
C02823__extr EX_C02823__dra 0.0001986        63  0.15%
C14818__extr EX_C14818__dra  0.004079         0  0.00%

Secretion
---------
      Metabolite            Reaction       Flux  C-Number  C-Flux
photon298__cytop DM_photon298__cytop      -0.14         0   0.00%
  photon438__lum   DM_photon438__lum     -20.32         0   0.00%
 photon450__chlo  DM_photon450__chlo     -17.23

In [7]:
dmodel.metabolites.C01389__chlo.summary(fva = 1.0)

In [16]:
from ExpGSMM.utils.utils import get_element_in_biomass

get_element_in_biomass(dmodel, "N", 'e_Biomass__cytop')

(1.0, {'C': 43.88272415658298, 'N': 2.616595001909996, 'O': 22.520446557234997, 'P': 0.16338696629600014, 'S': 0.112544626612, 'H': 65.92181810794798})
Error! Sum of Elemental mass percentage is different from 1! (0.999)


0.527

In [37]:
0.0309/60*1000

0.515

In [11]:
from ExpGSMM.utils.utils import get_element_in_biomass

get_element_in_biomass(dmodel, "P", 'e_ActiveBiomass__cytop')

(1.001, {'C': 45.31399797765508, 'N': 4.8696900171000586, 'O': 18.745534855713114, 'P': 0.2949523924470796, 'S': 0.20650366652204, 'H': 70.84365210840832})
Error! Sum of Elemental mass percentage is different from 1! (0.999)


0.068

In [18]:
4.18/100*Metabolite(formula='P').formula_weight

1.2947032098

In [19]:
0.12/100*Metabolite(formula='P').formula_weight

0.037168513199999996

In [30]:
1- 0.295 * (0.107688/0.137035) / 0.25

0.07270522129382995

In [33]:
dmodel.exchanges.EX_C00009__dra.bounds = (0, 1000)
dmodel.demands.DM_C00404__vacu.bounds = (-0.005, 1000)
dmodel.summary()

Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.
Could not identify an external compartment by name and choosing one with the most boundary reactions. That might be complete nonsense or change suddenly. Consider renaming your compartments using `Model.compartments` to fix this.


Runnning pFBA


In [45]:
dmodel.reactions.e_ActiveBiomass__cytop

<Reaction e_ActiveBiomass__cytop at 0x1dc1ba5a910>

In [66]:
dmodel.metabolites.C00009__cytop.summary()

In [60]:
dmodel.exchanges.EX_C00011__dra.lower_bound = -8.21
dmodel.demands.DM_C00369__chlo.bounds = (0.001219, 1000)
dmodel.demands.DM_C00116__cytop.bounds = (0.3252, 1000)
dmodel.demands.DM_C00422__lip.bounds = (0.002879, 1000)
dmodel.demands.DM_C02094__chlo.bounds = (0.0007998, 1000)
dmodel.demands.DM_C05306__chlo.bounds = (0.0001095, 1000)
dmodel.demands.DM_C05307__chlo.bounds = (3.683E-05, 1000)

In [62]:
sol = dmodel.optimize()
sol.objective_value

0.10707593093435432

In [59]:
dmodel.exchanges.EX_C00011__dra.lower_bound

-8.813

In [63]:
sol['e_ActiveBiomass__cytop'] + sol['DM_C00369__chlo']*48660.20/1000 + sol['DM_C00116__cytop']*92.09/1000 + sol['DM_C00422__lip']*904.78/1000 + sol['DM_C02094__chlo']*536.87/1000 + sol['DM_C05306__chlo']*893.49/1000 + sol['DM_C05307__chlo']*907.49/1000

0.19950589299205432

In [65]:
dmodel.summary()

Runnning pFBA


In [7]:
0.158*model.optimize().objective_value

0.030850355562195807

In [5]:
model.optimize()['EX_C00009__dra']

-0.030799368290332287

In [8]:
model.metabolites.C05307__chlo.summary()

In [8]:
model.summary()

Runnning pFBA


In [5]:
data_directory = "../data"
model = MyModel(join(data_directory, "models/model_with_trials.xml"), "e_Biomass__cytop")
model.add_medium(join(data_directory, "media.xlsx"), "base_medium")
model.setup_condition("default")

Loading

Reactions: 4591
Metabolites: 3658
Genes: 1690
Model loaded


In [6]:
model.metabolites.C02094__chlo.summary()

In [5]:
model.summary()

Runnning pFBA


In [32]:
model.reactions.DM_C00404__vacu.bounds = (-0.01, 1000)

In [34]:
model.test_reaction("R03042__vacu")

Flux
H2O            1500.000000
H+               99.382549
Polyphosphate     2.939357

In [36]:
fva(model, ['DM_C00404__vacu'], fraction_of_optimum=0.9)

minimum   maximum
DM_C00404__vacu -0.001315  1.559168

In [35]:
with model:
    model.exchanges.EX_C00205__dra.bounds = (-5000, 1000)
    model.exchanges.EX_C00011__dra.bounds = (-100, 1000)
    print(model.summary())

Runnning pFBA
Objective
1.0 e_Biomass_trial4__cytop = 1.7567534842773915

Uptake
------
  Metabolite       Reaction      Flux  C-Number C-Flux
C00001__extr EX_C00001__dra     59.12         0  0.00%
C00009__extr EX_C00009__dra     0.266         0  0.00%
C00011__extr EX_C00011__dra     72.03         1 99.77%
C00059__extr EX_C00059__dra    0.1616         0  0.00%
C00080__extr EX_C00080__dra    0.4334         0  0.00%
C00120__extr EX_C00120__dra  0.006345        10  0.09%
C00205__extr EX_C00205__dra      5000         0  0.00%
C00244__extr EX_C00244__dra     3.861         0  0.00%
C00305__extr EX_C00305__dra  0.000419         0  0.00%
C00378__extr EX_C00378__dra  0.003645        12  0.06%
C02823__extr EX_C02823__dra 0.0009812        63  0.09%
C14818__extr EX_C14818__dra   0.02017         0  0.00%

Secretion
---------
      Metabolite            Reaction       Flux  C-Number  C-Flux
photon298__cytop DM_photon298__cytop     -1.111         0   0.00%
  photon438__lum   DM_photon438__lum     -16

In [6]:
model.exchanges.EX_C00011__dra.bounds = (-1000, 1000)
model.exchanges.EX_C00205__dra.bounds = (-10000, 1000)
model.summary()

Runnning pFBA


In [35]:
model.metabolites.C02094__chlo.summary()

In [ ]:
model.metabolites.C00065__mito.summary(fva=1.0)

In [ ]:
with model:
    # model.create_sink("C00148__mito")
    print(fva(model, ['R02236__cytop'], fraction_of_optimum=0.0))

In [ ]:
model.test_reaction("R02236__cytop")

In [ ]:
model.metabolites.C00025__mito.summary(fva = 1.0)

In [ ]:
model.summary()

In [ ]:
#create copy reactions
with model:
    carbohydrate_copy = model.reactions.e_Carbohydrate__cytop.copy()
    carbohydrate_copy.id = 'e_Carbohydrate_no_starch__cytop'
    biomass_copy = model.reactions.e_Biomass__cytop.copy()
    biomass_copy.id = 'e_ActiveBiomass__cytop'
    model.add_reactions([carbohydrate_copy, biomass_copy])
    model.objective = 'e_ActiveBiomass__cytop'
    model.reactions.e_Carbohydrate__cytop.bounds = (0, 0)
    #remove starch from e_carbohydrate
    carbohydrate_copy.add_metabolites({model.metabolites.C00369__chlo: -carbohydrate_copy.metabolites[model.metabolites.C00369__chlo]})

    #adjust other carbohydrates
    carbohydrate_copy.add_metabolites({model.metabolites.C00052__cytop:  -3.35762291 - carbohydrate_copy.metabolites[model.metabolites.C00052__cytop]})
    carbohydrate_copy.add_metabolites({model.metabolites.C00096__cytop:  -1.58181346 - carbohydrate_copy.metabolites[model.metabolites.C00096__cytop]})
    carbohydrate_copy.add_metabolites({model.metabolites.C00935__cytop:  -2.52493299 - carbohydrate_copy.metabolites[model.metabolites.C00935__cytop]})

    carbohydrate_copy.add_metabolites({model.metabolites.C00015__cytop:  5.8825559 - carbohydrate_copy.metabolites[model.metabolites.C00015__cytop]})
    carbohydrate_copy.add_metabolites({model.metabolites.C00035__cytop:  1.58181346 - carbohydrate_copy.metabolites[model.metabolites.C00035__cytop]})
    carbohydrate_copy.add_metabolites({model.metabolites.C00001__cytop:  7.46436936 - carbohydrate_copy.metabolites[model.metabolites.C00001__cytop]})

    print(carbohydrate_copy.metabolites)

    biomass_copy.add_metabolites({model.metabolites.e_Carbohydrate__cytop:  -0.203306526 - biomass_copy.metabolites[model.metabolites.e_Carbohydrate__cytop]})
    print(biomass_copy.metabolites)
    print(model.optimize().objective_value*0.7083)

    # total = abs(sum([biomass_copy.metabolites[met] for met in biomass_copy.reactants if biomass_copy.metabolites[met] > -10]))
    # for  met in biomass_copy.reactants:
    #     if biomass_copy.metabolites[met] > -10:
    #         print(met.id)
    #         new_st = biomass_copy.metabolites[met]/total
    #         print(new_st)
    #         print(biomass_copy.metabolites[met])
    #         biomass_copy.add_metabolites({model.metabolites.get_by_id(met.id):  new_st - biomass_copy.metabolites[met]})
    #         print(biomass_copy.metabolites[met])
    # print(sum([biomass_copy.metabolites[met] for met in biomass_copy.reactants if biomass_copy.metabolites[met] > -10]))
    # print(model.optimize().objective_value)
    model.write(os.path.join(data_directory, "models/model_dfba.xml"))

In [ ]:
data_directory = "../data"
model = MyModel(join(data_directory, "model_dfba.xml"), "e_Biomass_trial4__cytop")
model.add_medium(join(data_directory, "media.xlsx"), "base_medium")
model.setup_condition("4")

In [ ]:
pfba_sol = pfba(model)
model.summary(pfba_sol)

In [ ]:
with model:
    model.reactions.DM_C00369__chlo.bounds = (0.00147, 1000)
    pfba_sol = pfba(model)
    # print(model.metabolites.C00369__chlo.summary(pfba_sol))
    print(model.summary(pfba_sol))

In [ ]:
sum([model.reactions.e_Biomass_trial4__cytop.metabolites[reactant] for reactant in model.reactions.e_Biomass_trial4__cytop.metabolites if model.reactions.e_Biomass_trial4__cytop.metabolites[reactant] < 0 and model.reactions.e_Biomass_trial4__cytop.metabolites[reactant]>-1])

In [ ]:
original_sol = model.optimize().objective_value

In [ ]:
model.summary()

In [ ]:
model.metabolites.C00116__cytop.formula_weight

In [ ]:
fva(model, ["T_SUCBZth__chlomem"])

In [ ]:
with model:
    # model.biomass_reaction = model.bio_reaction.id
    # model.reactions.e_Carbohydrate__cytop.add_metabolites({model.metabolites.C00369__chlo: -model.reactions.e_Carbohydrate__cytop.metabolites[model.metabolites.C00369__chlo]})
    # model.reactions.e_Biomass_trial4__cytop
    # model.reactions.e_Acid__cytop.add_metabolites({model.metabolites.C00116__cytop: -model.reactions.e_Acid__cytop.metabolites[model.metabolites.C00116__cytop]})
    # model.reactions.DM_C00369__chlo.bounds = (0.305381773/model.metabolites.C00369__chlo.formula_weight*1000*original_sol, 1000)
    # model.reactions.DM_C00369__chlo.bounds = (0.00147, 1000)
    # [setattr(x, 'objective_coefficient', 0) for x in model.reactions if x.objective_coefficient != 0]
    # model.reactions.EX_e_Biomass__dra.objective_coefficient = 0.7
    # model.reactions.DM_C00369__chlo.objective_coefficient = 0.3
    # model.objective = "e_Biomass__cytop"
    sol = pfba(model)
    print(model.summary(solution=sol))
    # print(model.metabolites.C00369__chlo.summary(sol))

In [ ]:
total = sum([reactant for reactant in model.reactions.e_Biomass__cytop.metabolites if model.reactions.e_Biomass__cytop.metabolites[reactant] < 0 and model.reactions.e_Biomass__cytop.metabolites[reactant]>-1])
for reactant in model.reactions.e_Biomass__cytop.reactants:
    st =

model.reactions.DM_C00369__chlo

model.reactions.e_Lipid__cytop.add_metabolites({model.metabolites.C00422__lip: 0.13})

model.reactions.e_Lipid__cytop.metabolites

model.reactions.e_Carbohydrate__cytop.metabolites

0.3212*(1-(0.0123373*48660.08/1000*0.50869+0.13*904.7821/1000*0.11407))

(1-(0.0123373*48660.08/1000*0.50869+0.13*904.7821/1000*0.11407))

model.reactions.DM_C00369__chlo.bounds = (0.3/48660.08*1000*0.21, 1000)

model.reactions.DM_C00369__chlo

model.summary()